browser/ui/webui/brave_rewards_ui.cc

```diff
@@ -610,7 +610,22 @@ void RewardsDOMHandler::SaveSetting(const base::ListValue* args) {
     args->GetString(1, &value);
 
     if (key == "enabledMain") {
-      rewards_service_->SetRewardsMainEnabled(value == "true");
+      bool rewards_enabled = (value == "true");
+
+      if (ads_service_) {
+        if (rewards_enabled) {
+          // When Rewards are enabled, set Ads enabled to whatever UI value is
+          bool ads_ui_enabled = ads_service_->is_ui_enabled();
```

Comment:
I think it's better to use the buildflag for this. I would only use the `ui_enabled` stuff in js

Reply:
or is this something different? I guess I'm confused

Reply:
what is the difference between `enabled` and `ui_enabled`?

Reply:
actually I think I understand, but per the comment below I think ads service should have logic to check the status of rewards service `enabled`. Per discussions with @tmancey ads are going to be more tightly integrated inside the service process

Reply:
So `enabled` essentially means, "is the ad service enabled and running?" `ui_enabled` means, "have ads been explicitly enabled in the UI by the user?"

This may make it more clear

User has Rewards enabled via the main toggle, and the ads box enabled:
`enabled` -> `true`
`ui_enabled` -> `true`

The user then disables the main Rewards toggle
`enabled` -> `false` (Ads service is off as a result of shutting off rewards)
`ui_enabled` -> `true` (This is unchanged as just Rewards was disabled, not explicitly Ads)

With the above when one re-enables Rewards via the main toggle, we have the preference for Ads being on saved in `ui_enabled`, so we know we can turn Ads back on along with Rewards

Reply:
I think if ads is dependent on rewards then it makes sense to have AdsServiceFactory DependsOn for Rewards to make it explicit and keep the enable/disable logic in the service

Reply:
this is what I wanted to do as well (https://github.com/brave/brave-browser/issues/2888#issuecomment-452987399), but because of time constrains we just added new pefr for now so that we could ship it today